src/Symfony/Component/Workflow/Tests/StateMachineTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Symfony\Component\Workflow\Tests;
+
+use Symfony\Component\Workflow\Definition;
+use Symfony\Component\Workflow\Marking;
+use Symfony\Component\Workflow\StateMachine;
+use Symfony\Component\Workflow\Transition;
+
+class StateMachineTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @expectedException \Symfony\Component\Workflow\Exception\LogicException
+     */
+    public function testConstructorWithMultipleTransitionWithSameNameShareInput()
+    {
+        $places = array('a', 'b', 'c');
+        $transitions[] = new Transition('t1', 'a', 'b');
+        $transitions[] = new Transition('t1', 'a', 'c');
+        $definition = new Definition($places, $transitions);
+
+        $net = new StateMachine($definition);
+    }
+
+    public function testCanWithStateMachineMarkingStore()
+    {
+        $places = array('a', 'b', 'c', 'd');
+        $transitions[] = new Transition('t1', 'a', 'b');
+        $transitions[] = new Transition('t1', 'd', 'b');
+        $transitions[] = new Transition('t2', 'b', 'c');
+        $transitions[] = new Transition('t3', 'b', 'd');
+        $definition = new Definition($places, $transitions);
+
+        $net = new StateMachine($definition);
+        $subject = new \stdClass();
+
+        // If you are in place a you should be able to apply t1
```

Comment:
`// If you are in place "a" you should be able to apply "t1"`
(same comment for other occurrence)